Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so and /home/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so differ
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so.0 and /home/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so.0 differ
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so.0.0.0 and /home/donald/wxWindows/wxGTK-2.4.1/lib/libwx_gtkd-2.4.so.0.0.0 differ
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/listctrl.o and /home/donald/wxWindows/wxGTK-2.4.1/listctrl.o differ
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/list.o and /home/donald/wxWindows/wxGTK-2.4.1/list.o differ
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest and /home/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest differ
diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' /mnt/temp/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.d /home/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.d
--- /mnt/temp/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.d	2003-06-29 12:39:43.000000000 -0500
+++ /home/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.d	2003-06-30 12:55:22.000000000 -0500
@@ -1,5 +1,5 @@
 listtest.o: listtest.cpp ../../include/wx/wxprec.h \
   ../../include/wx/defs.h ../../include/wx/platform.h \
-  ../../lib/wx/include/gtk-2.4/wx/setup.h ../../include/wx/chkconf.h \
+  ../../lib/wx/include/gtkd-2.4/wx/setup.h ../../include/wx/chkconf.h \
   ../../include/wx/features.h ../../include/wx/version.h \
   ../../include/wx/debug.h ../../include/wx/wxchar.h \
Binary files /mnt/temp/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.o and /home/donald/wxWindows/wxGTK-2.4.1/samples/listctrl/listtest.o differ
diff -ru2 -xCVS -x'.#*' -x'*.orig' -x'*.rej' -x'*~' /mnt/temp/donald/wxWindows/wxGTK-2.4.1/src/generic/listctrl.cpp /home/donald/wxWindows/wxGTK-2.4.1/src/generic/listctrl.cpp
--- /mnt/temp/donald/wxWindows/wxGTK-2.4.1/src/generic/listctrl.cpp	2003-06-08 03:13:16.000000000 -0500
+++ /home/donald/wxWindows/wxGTK-2.4.1/src/generic/listctrl.cpp	2003-06-30 17:29:56.000000000 -0500
@@ -4289,4 +4289,13 @@
     m_columns.DeleteNode( node );
 
+    //dtaylor added to allow remove column from rows of data
+    for(wxUint16 i=0; i<m_lines.GetCount(); i++)
+    {
+        wxListLineData *line = GetLine(i);
+        wxListItemDataList::Node *n = line->m_items.Item( col );
+        line->m_items.DeleteNode(n);
+    }
+    // end of modification
+
     // invalidate it as it has to be recalculated
     m_headerWidth = 0;
@@ -4486,12 +4495,27 @@
             wxListHeaderDataList::Node *node = m_columns.Item( col );
             m_columns.Insert( node, column );
+            //dtaylor added to allow row of data to add a new column dynamically (06/30/2003)
+            for(wxUint16 i=0; i<m_lines.GetCount(); i++)
+            {
+                wxListLineData *line = GetLine(i);
+                line->m_items.Insert(col,new wxListItemData(line->m_owner));
+            }
+            // end of modification ****
         }
         else
         {
             m_columns.Append( column );
+            //dtaylor added to allow row of data to add a new column dynamically (06/30/2003)
+            for(wxUint16 i=0; i<m_lines.GetCount(); i++)
+            {
+                wxListLineData *line = GetLine(i);
+                line->m_items.Append(new wxListItemData(line->m_owner));
+            }
+            // end of modification
         }
 
         // invalidate it as it has to be recalculated
         m_headerWidth = 0;
+
     }
 }

 	  	 
